### ISSUE TITLE
Adding more scientific context to USID Main datasets

### ISSUE THREAD

**Issue by ssomnath** [pycroscopy/USID#3]
At the moment, the only mandatory attributes that must be written to Main datasets are `quantity` and `units`. It may be of value to analytics packages (such as pyUSID) to identify the dataset as a `PFM Imaging` or `BEPS` or `cKPFM` dataset and not just any USID Main dataset. Perhaps the presence of such keywords or flags could facilitate the development of children classes of pyUSID.USIDataset that add more value to the slicing, analysis, visualization capabilities.

**Comment by ramav87** [comment]
Agreed, this would be especially useful for the more complex spectroscopies like relaxation or time-resolve KPFM. @Liambcollins we should discuss the common threads here.
